### PR TITLE
Snowbridge - Copy Rococo integration tests to Westend

### PR DESCRIPTION
**Pull request by claravanstaden**
Copies all the integration tests from Rococo to Westend.

Closes: https://github.com/paritytech/polkadot-sdk/issues/6389

### CODE REVIEW

**Comment by claravanstaden** [comment]
Requires labels: [R0-silent](https://github.com/paritytech/polkadot-sdk/pulls?q=is%3Apr+is%3Aopen+label%3AR0-silent) and [T15-bridges](https://github.com/paritytech/polkadot-sdk/pulls?q=is%3Apr+is%3Aopen+label%3AT15-bridges)

**Comment by claravanstaden** [comment]
@yrong @acatangiu please review. :)

File: cumulus/parachains/integration-tests/emulated/tests/bridges/bridge-hub-westend/src/tests/snowbridge.rs [hunk 1]
```diff
@@ -12,35 +12,46 @@
...
 
 #[derive(Encode, Decode, Debug, PartialEq, Eq, Clone, TypeInfo)]
 pub enum ControlCall {
 	#[codec(index = 3)]
 	CreateAgent,
 	#[codec(index = 4)]
-	CreateChannel { mode: OperatingMode },
+	CreateChannel {
+		mode: OperatingMode,
+	},
```
Comment (yrong) [inline cumulus/parachains/integration-tests/emulated/tests/bridges/bridge-hub-westend/src/tests/snowbridge.rs:53]: May not need `CreateChannel` anymore?

File: cumulus/parachains/integration-tests/emulated/tests/bridges/bridge-hub-westend/src/tests/snowbridge.rs [hunk 2]
```diff
@@ -12,35 +12,46 @@
...
+	Pe,
```
Comment (yrong) [inline cumulus/parachains/integration-tests/emulated/tests/bridges/bridge-hub-westend/src/tests/snowbridge.rs:54]: just curious what is this for?

File: cumulus/parachains/integration-tests/emulated/tests/bridges/bridge-hub-westend/src/tests/snowbridge.rs [hunk 3]
```diff
@@ -82,6 +240,566 @@ fn register_weth_token_from_ethereum_to_asset_hub() {
...
+			AssetHubWestend,
+			vec![
+				RuntimeEvent::ForeignAssets(pallet_assets::Event::Issued { .. }) => {},
+			]
+		);
+	});
+}
+
+#[test]
+fn send_weth_from_ethereum_to_non_existent_account_on_asset_hub() {
```
Comment (yrong) [inline cumulus/parachains/integration-tests/emulated/tests/bridges/bridge-hub-westend/src/tests/snowbridge.rs:750]: I'd suggest adding these edge cases to a single file.

File: cumulus/parachains/integration-tests/emulated/tests/bridges/bridge-hub-westend/src/tests/snowbridge.rs (hunk 1 above)
Comment (claravanstaden) [reply cumulus/parachains/integration-tests/emulated/tests/bridges/bridge-hub-westend/src/tests/snowbridge.rs]: Good catch! https://github.com/paritytech/polkadot-sdk/pull/7108/commits/d7dee7f53358ae4b4a4fdfdffb331fc686ef2e17

File: cumulus/parachains/integration-tests/emulated/tests/bridges/bridge-hub-westend/src/tests/snowbridge.rs (hunk 2 above)
Comment (claravanstaden) [reply cumulus/parachains/integration-tests/emulated/tests/bridges/bridge-hub-westend/src/tests/snowbridge.rs]: I am not sure either 😂 I think I started adding something related to Penpal, then it wasn't necessary and I forgot to remove it.
https://github.com/paritytech/polkadot-sdk/pull/7108/commits/d7dee7f53358ae4b4a4fdfdffb331fc686ef2e17

File: cumulus/parachains/integration-tests/emulated/tests/bridges/bridge-hub-westend/src/tests/snowbridge.rs (hunk 3 above)
Comment (claravanstaden) [reply cumulus/parachains/integration-tests/emulated/tests/bridges/bridge-hub-westend/src/tests/snowbridge.rs:667]: Could you elaborate why? I don't think it is necessary.

**Review by yrong — APPROVED** [review]
+1

**Review by acatangiu — APPROVED** [review]

**Comment by acatangiu** [comment]
tests need fixing

**Review by bkontur — APPROVED** [review]
